src/DNA/corlib/System.Collections.Generic/IReadOnlyCollection.cs

```diff
@@ -0,0 +1,33 @@
+//
+// IReadOnlyCollection.cs
```

Comment:
Why copy from mono instead of corefx?

Reply:
No particular reason, this interface is the same either way. Perhaps one reason, the original DNA seems to be based on mono (although I can't be 100% sure), and the mono implementation is usually a bit smaller and simpler. In any case, I have refrained from copying too much, only the bare essentials to get things going.

Reply:
Yes, DNA originally used Mono code for classes that didn't require any integration into the C interpreter, and I didn't have the time/inclination to write from scratch. Clearly corefx didn't exist at the time!
Note that the classes I wrote were generally written to be small, rather than performant; the platform I originally wrote this for was *very* resource limited. Although this was all so long ago I can't really remember details.
Switching to using corefx code might now make some sense.